src/libs/MigrateOnyx.js

```diff
@@ -0,0 +1,22 @@
+import promiseAllSettled from './promiseAllSettled';
+import RenameActiveClientsKey from './migrations/RenameActiveClientsKey';
+
+export default function () {
+    const startTime = Date.now();
+    console.debug('[Migrate Onyx] start');
+
+    return new Promise((resolve) => {
+        // Add all migrations to an array and execute them, they can execute in any order
```

Comment:
I'm curious why they should they be able to execute in any order? What if you need  them to happen in a particular order? e.g.

**migration on monday:**
split complex object into two keys

**migration on tuesday:**
rename one of those keys

Reply:
While being sensitive to not over-engineering this, I don't think there is a reason to do this today. Another benefit is that it will be much faster if every migration can be run in parallel and in any order (rather than being done in-sequence) which also follows the other patterns we are trying to put forth in this code.

In your case above, I think when you add the second migration, then it should:
1. Write the Tuesday migration to split the complex object into two keys with the proper name
2. Look for the wrong key name created from Monday and migrate them to the new name
3. Remove the Monday migration

Reply:
That makes sense to me, but to flip it around ... perhaps we are over-engineering this already by assuming linear migrations will be slow and running things in parallel is more desirable?

The counter proposal there is interesting, but I'm confused about why that would be preferable to just having a series of linear migrations that run in sequence to take some state to another state and have a clear story about when they have changed by applying one after the next.

If we are allowing migrations to be "edited" after the fact won't we need to take special care to not assume the state that someone's app is in? Linear migrations seem safer and give us less to worry about. As long as you're adding a new migration to run after a previous one then you can always assume that the data you need can be accessed in the next migration in the sequence.

Maybe I'm off base here since I haven't tried to solve this problem before. Curious if @iwiznia has thoughts on this.

Reply:
> why that would be preferable to just having a series of linear migrations that run in sequence

Well, I think that speed and performance for something that blocks the app from loading are absolutely necessary. Running 4 things in sequence is always going to take longer than doing 4 things in parallel (unless I'm confused on what you're suggesting).

> Linear migrations seem safer and give us less to worry about

I understand that there might be some uncertainty here, but let's be sure we're fixing real problems as they happen. It would be best to lead with the most simple approach and then refactor it later if we find it's not desireable.

Reply:
🐰